### PR TITLE
Fix central manage login and logout bugs.

### PR DESCRIPTION
**Pull request by albarnaz**
[#18160](https://central.cm42.io/projects/central#story-18160)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -56,9 +56,8 @@ def current_team
   helper_method :current_team
 
   def after_sign_in_path_for(resource)
-    if resource.authy_enabled && resource.authy_id.blank?
-      return send("#{resource_name}_enable_authy_path")
-    end
+    return super if resource.is_a?(AdminUser)
+    return send("#{resource_name}_enable_authy_path") if resource.authy_enabled && resource.authy_id.blank?
```
Comment (tiarly) [inline app/controllers/application_controller.rb]: let's use `public_send` here instead

Comment (tiarly) [inline app/controllers/application_controller.rb]: let's use `public_send` here instead

**Comment by tiarly** [comment]
Shouldn't also add another spec on `spec/controllers/admin/users_controller_spec.rb` with our `UserAdmin` use case? This way we can guarantee we won't have any regressions on this anymore

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (tiarly) [inline app/controllers/application_controller.rb]: We could also move this method to its on class, so we can't better unit test this behavior 💅

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## [Unreleased]
+Fix central manage login and logout bugs
```
Comment (andrerpbts) [inline CHANGELOG.md]: Add it into a `Fixed` group.

You can obtain more info [here](http://keepachangelog.com/en/1.0.0/)

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ class ApplicationController < ActionController::Base
   include SidebarController
 
   before_action :authenticate_user!, unless: :devise_controller?
-  before_action :check_team_presence, unless: :devise_controller?
+  before_action :check_team_presence, if: -> { current_user.present? }
```
Comment (andrerpbts) [inline app/controllers/application_controller.rb]: well, do not discard the previous condition...

In this case you should check both if is a devise_controller and if current_user is present.

File: config/initializers/active_admin.rb [hunk 4]
```diff
@@ -34,6 +34,7 @@
   #
   # Default:
   config.default_namespace = :manage
+  # config.skip_before_filter = :authenticate_user!
```
Comment (andrerpbts) [inline config/initializers/active_admin.rb]: IMO this is a necessary behaviour. If this method is failing, try to use `skip_before_action` instead.

Also, owrking here, you can remove this line from the monkey patch in this same file below.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (andrerpbts) [reply app/controllers/application_controller.rb]: Also, this change will help to solve the codeclimate issue aswell...

**Pull request merged**